compiler.ts

```diff
@@ -305,6 +496,66 @@ function codeGenExpr(expr: Expr<Type>, env: GlobalEnv): Array<string> {
       var className = objTyp.name;
       var [offset, _] = env.classes.get(className).get(expr.field);
       return [...objStmts, `(i32.add (i32.const ${offset * 4}))`, `(i32.load)`];
+    case "bracket-lookup":
+      var objStmts = codeGenExpr(expr.obj, env);
+      //This should eval to a number
+      //Multiply it by 4 to use as offset in memory
+      var keyStmts = codeGenExpr(expr.key, env);
```

Comment:
Yes, this would be problematic because the result of evaluating the keyStmts will have a tag bit in the LSB. Our pull request will provide a helper that removes the tag bit